File: aat-lib/src/main/java/ch/bailu/aat_lib/xml/parser/gpx/GpxExtensionParser.java
package ch.bailu.aat_lib.xml.parser.gpx;

import org.xmlpull.v1.XmlPullParser;

import ch.bailu.aat_lib.xml.parser.osm.TagParser;
import ch.bailu.aat_lib.xml.parser.scanner.Scanner;

public class GpxExtensionParser extends TagParser {


    private String key = null;


    public GpxExtensionParser() {
        super(null);
    }


    @Override
    protected void parseText(XmlPullParser parser, Scanner scanner) {
        final String val = parser.getText();
        if (val != null && key != null && key.length() > 0) {
            scanner.tags.add(key, val);

        }
    }


    @Override
    protected void parseAttributes(XmlPullParser parser, Scanner scanner) {
        key = parser.getName();

    }

    @Override
    protected boolean parseTags(XmlPullParser parser, Scanner scanner) {
        return false;
    }

    @Override
    protected void parsed(XmlPullParser parser, Scanner scanner) {

    }
}

File: aat-lib/src/main/java/ch/bailu/aat_lib/file/xml/parser/gpx/GpxExtensionParser.kt
package ch.bailu.aat_lib.file.xml.parser.gpx

import ch.bailu.aat_lib.file.xml.parser.osm.TagParser
import ch.bailu.aat_lib.file.xml.parser.scanner.Scanner
import org.xmlpull.v1.XmlPullParser

class GpxExtensionParser : TagParser() {
    private var key: String = ""

    override fun parseText(parser: XmlPullParser, scanner: Scanner) {
        val value = parser.text
        if (value != null && key.isNotEmpty()) {
            scanner.tags.add(key, value)
        }
    }

    override fun parseAttributes(parser: XmlPullParser, scanner: Scanner) {
        key = parser.name
    }

    override fun parseTags(parser: XmlPullParser, scanner: Scanner): Boolean {
        return false
    }

    override fun parsed(parser: XmlPullParser, scanner: Scanner) {}
}
